cns/restserver/ipam.go

```diff
@@ -1136,25 +1136,39 @@ func (service *HTTPRestService) UpdateEndpointHelper(endpointID string, req map[
 	}
 	logger.Printf("[updateEndpoint] Updating endpoint state for infra container %s", endpointID)
 	if endpointInfo, ok := service.EndpointState[endpointID]; ok {
-		for ifName, InterfaceInfo := range req {
-			logger.Printf("[updateEndpoint] Found existing endpoint state for infra container %s", endpointID)
-			if InterfaceInfo.HnsEndpointID != "" {
-				service.EndpointState[endpointID].IfnameToIPMap[ifName].HnsEndpointID = InterfaceInfo.HnsEndpointID
-				logger.Printf("[updateEndpoint] update the endpoint %s with HNSID  %s", endpointID, InterfaceInfo.HnsEndpointID)
+		for ifName, interfaceInfo := range req {
+			// This codition will create a map for SecodaryNIC and also also creates MAP entry for InfraNic in case that the initial goalState is using empty InterfaceName
+			if _, keyExist := service.EndpointState[endpointID].IfnameToIPMap[ifName]; !keyExist {
```

Comment:
I think this needs some commenting. There's a lot of complex map manipulation (types in maps of maps) happening here that is not obvious to readers. Either that, or it might be more helpful to encapsulate what's happening here in a higher-level type whose operations are more intention-revealing.

Reply:
A comment has been made on top of the for block

Reply:
this is still very complicated to read. I agree with @timraymond , we should do some encapsulation here.

Reply:
@timraymond @nddq @tamilmani1989 
Added a new commit addressing the issue with a new func that encapsulates and simplified the block